projects/ngx-prx-styleguide/src/lib/datepicker/datepicker.component.ts

```diff
@@ -94,11 +94,11 @@ export class DatepickerComponent implements AfterViewInit {
     if (moment(value, this.format, true).isValid() &&
       (!this._date || this.picker.getDate().valueOf() !== this._date.valueOf())) {
       let date = new Date(value);
+      this.setDate(date);
```

Comment:
I believe that moving the call to `setDate` (and thus emitting `dateChange`) before updating the picker won't get us into an infinite update loop, as there's a guard clause in the date setter with a call to `dateDefinedAndChanged`.

This resolves the double offset bug in the previous implementation:

> This will create a new date in the user’s locale regardless of the UTC setting, but then the picker date is still being offset to compensate if UTC is true. This results in the date being 12 hours behind in my testing (GMT-6 for my timezone plus another -6 added for the offset)